### PR TITLE
feat(profile): allow to delete a validator profile

### LINKED ISSUE #189: profile: allow to delete a validator profile
Implement a message that will delete the validator profile entry from the store

### PR DESCRIPTION
**Pull request by Pantani**
Closes #189

### What does this PR does?

- Create `delete-validator` structure;
- Implemente `DeleteValidator` method;
- Add `DeleteValidator` tests (`TestMsgDeleteValidator`);

### CODE REVIEW

File: x/profile/keeper/msg_server_delete_validator.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func (k msgServer) DeleteValidator(goCtx context.Context, msg *types.MsgDeleteValidator) (*types.MsgDeleteValidatorResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the validator address is already in the store
+	_, found := k.GetValidator(ctx, msg.Address)
+	if !found {
+		return &types.MsgDeleteValidatorResponse{},
+			sdkerrors.Wrap(types.ErrValidatorNotFound,
```
Comment (lubtd) [inline x/profile/keeper/msg_server_delete_validator.go:19]: `sdkerrors` has a `Wrapf` method when the string is formated

File: x/profile/keeper/msg_server_delete_validator.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+				fmt.Sprintf("validator: %s", msg.Address))
```
Comment (lubtd) [inline x/profile/keeper/msg_server_delete_validator.go:20]: I think just precising the address is sufficient based on the wrapped error

File: x/profile/types/message_delete_validator.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+func (msg *MsgDeleteValidator) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgDeleteValidator) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address address (%s)", err)
```
Comment (lubtd) [inline x/profile/types/message_delete_validator.go:40]:
```suggestion
		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid validator address (%s)", err)
```

File: x/profile/keeper/msg_server_delete_validator.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package keeper
+
+import (
+	"context"
```
Comment (lubtd) [inline x/profile/keeper/msg_server_delete_validator.go:4]:
```suggestion
	"context"
	
```

**Review by lubtd — COMMENTED** [review]
Code looks legit to me👍

We will need https://github.com/tendermint/spn/pull/198 to be merged first in order to correctly test the feature from the CLI

**Comment by lubtd** [comment]
We will actually have to change the specs a bit.

We should let the possibility to the coordinator to remove an account or a validator in case it has been mistakenly approved and some data are wrong (some checks must be performed off-chain for the validator).

We need to think about two things:
- Does anyone could send a removal request or either the account itself or the coordinator? In bother case, we will need to add a new field for the message

- Should coordinator request be automatically pre approved? This is something we had in the legacy version although this detail is not mentioned in the specs. We will first need to implement `MsgSettleRequest` for this so this doesn't influence this PR

**Comment by Pantani** [comment]
> We will actually have to change the specs a bit.
> 
> We should let the possibility to the coordinator to remove an account or a validator in case it has been mistakenly approved and some data are wrong (some checks must be performed off-chain for the validator).
> 
> We need to think about two things:
> 
> * Does anyone could send a removal request or either the account itself or the coordinator? In bother case, we will need to add a new field for the message
> * Should coordinator request be automatically pre approved? This is something we had in the legacy version although this detail is not mentioned in the specs. We will first need to implement `MsgSettleRequest` for this so this doesn't influence this PR

If I understand the concept right, the coordinator sends the removal can be better for blockchain itself. He can drop peers with too much downtime or malicious. I think in most cases, the validators just shut down your nodes instead send a removal message before the shutdown.

IMHO, let's keep the automatically pre-approved by the coordinator, and we can add it to the specs

**Comment by lubtd** [comment]
Seems like I put my previous comment above on the wrong PR. It should be here: https://github.com/tendermint/spn/pull/216

**Review by lubtd — APPROVED** [review]

**Review by ilgooz — APPROVED** [review]

**Pull request merged**